### PR TITLE
Nexus 4918

### PR DESCRIPTION
**Pull request by velo**
Found a bug on metadata maintenance while testing staging SNAPSHOT

### CODE REVIEW

**Comment by nabcos** [comment]
+1

File: nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/maven/DefaultMetadataUpdater.java [hunk 1]
```diff
@@ -39,22 +40,29 @@
...
+    }
 
     public void deployArtifact( ArtifactStoreRequest request )
         throws IOException
     {
-        if ( request.getGav().isHash() || request.getGav().isSignature()
-            || StringUtils.isNotBlank( request.getGav().getClassifier() ) )
+        final Gav requestGav = request.getGav();
+        if ( requestGav.isHash() || requestGav.isSignature()
+            || ( StringUtils.isNotBlank( requestGav.getClassifier() ) && !requestGav.isSnapshot() ) )
```
Comment (mcculls) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/maven/DefaultMetadataUpdater.java]: I think undeployArtifact also needs fixing in the same way

Comment (velo) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/maven/DefaultMetadataUpdater.java]: Nice catch!

**Comment by mcculls** [comment]
+1

**Pull request merged**